### ISSUE TITLE
How to set "Item Type" property to "Copy file"?

### ISSUE THREAD

**Issue by mfloriani** [premake/premake-core#1677]
Hey,

Can I set the `Item Type` property of a file in the project to `Copy file` using premake?
I'm using VS2019 in a C++17 project.

![image](https://user-images.githubusercontent.com/13739810/127479620-1084c3ad-9517-4696-8354-d65926d7d591.png)

**Comment by starkos** [comment]
You would use [buildaction](https://premake.github.io/docs/buildaction/) for that.

```lua
filter { 'files:**.txt' }
   buildaction 'Copy'
```

This might not be supported for all exporters yet, but pretty sure it should work with VS 2019.

**Comment by nickclark2016** [comment]
> 
> 
> You would use [buildaction](https://premake.github.io/docs/buildaction/) for that.
> 
> ```lua
> filter { 'files:**.txt' }
>    buildaction 'Copy'
> ```
> 
> This might not be supported for all exporters yet, but pretty sure it should work with VS 2019.

Is there somewhere that can be looked at to find all actions for C++?

**Comment by starkos** [comment]
> Is there somewhere that can be looked at to find all actions for C++?

There does not seem to be. Here's everything I was able to find in sources…

- Embed
- Compile
- Copy
- Component
- Form
- UserControl
- Resource
- Application
- FxCompile
- None

**Comment by nickclark2016** [comment]
Hmm, looking at the documentation of Premake, it makes it sound like those are only available for C# projects.

**Comment by samsinsane** [comment]
Correct, those are the C# values. The C++ values are the element names and they're not validated. So, instead of `Compile` like C#, it's `ClCompile` for C++. Handling C++ that way was probably a mistake, and maybe we should consider mapping the C# values to C++ - everyone seems to get confused by the documentation anyway.
![image](https://user-images.githubusercontent.com/11887457/128882246-377eb47e-748f-4c95-a04b-748a4352b7a2.png)
